### ISSUE TITLE
[PRO] startCursor and endCursor are null after upgrade when using GraphQL::Pro::MySQLStableRelationConnection

### ISSUE THREAD

**Issue by gastonmorixe** [rmosolgo/graphql-ruby#3752]
We are trying to upgrade to v1.13.0 and pro v1.20.3 and a few of our tests fail. 

Just found that `startCursor` and `endCursor` are always null. They are ok if we comment this line 

```ruby
  # connections.add(ActiveRecord::Relation, GraphQL::Pro::MySQLStableRelationConnection)
```

**Comment by gastonmorixe** [comment]
I can't see pro code changes since it's private, but something is going on maybe around here:  `@nodes` is nil before calling `super`, after calling `super`  `@nodes` is correctly set and the cursor is ok. 

The issue seems to be to me that condition that short circuits first at `@nodes` being nil 

```
[66, 75] in /Users/gastonmorixe/.rvm/gems/ruby-2.7.3/gems/graphql-pro-1.20.3/lib/graphql/pro/stable_relation_connection.rb
   66:         !!@has_next_page
   67:       end
   68:
   69:       def start_cursor
   70:         debugger
=> 71:         @nodes && super
   72:       end
   73:
   74:       def end_cursor
   75:         debugger
(byebug) @nodes && super
nil
(byebug)  @nodes
nil
(byebug) super
...
"WyIyMDIxLTEyLTAzIDE2OjUxOjU0LjAwMDAwMDAwMCBVVEMiLDRd"
(byebug)  @nodes
[#<SmsConversationMessage id: 146, sms_conversation_id ....
```

**Comment by rmosolgo** [comment]
Thanks for the detailed report! It looks like I messed it up when I was trying to fix handling of invalid cursors in #3735, and I didn't have a test for fetching cursors without nodes. I added a test and it failed just like you said, and I released a fix in 1.20.4. Please give it a try and let me know if you run into any more trouble! 

In case it helps in the future, you can always browse the graphql-pro source code with `bundle open graphql-pro`. The license allows archival copies too, so you can duplicate it and diff versions against each other.

**Comment by gastonmorixe** [comment]
working fine now, thank you!
